DuckDuckGo/SiteRatingView.swift

```diff
@@ -38,27 +38,19 @@ public class SiteRatingView: UIView {
         let view = Bundle.main.loadNibNamed("SiteRatingView", owner: self, options: nil)![0] as! UIView
         self.addSubview(view)
         view.frame = self.bounds
-        addUserDefaultsObserver()
-    }
-    
-    deinit {
-        removeUserDefaultsObserver()
+        addContentBlockerConfigurationObserver()
     }
     
     public override func layoutSubviews() {
         super.layoutSubviews()
         refresh()
     }
-    
-    private func addUserDefaultsObserver() {
-        NotificationCenter.default.addObserver(self, selector: #selector(onUserDefaultsChanged), name: UserDefaults.didChangeNotification, object: nil)
-    }
-    
-    private func removeUserDefaultsObserver() {
-        NotificationCenter.default.removeObserver(self, name: UserDefaults.didChangeNotification, object: nil)
+
+    private func addContentBlockerConfigurationObserver() {
+        NotificationCenter.default.addObserver(self, selector: #selector(onContentBlockerConfigurationChanged), name: ContentBlockerConfigurationChangedNotification.name, object: nil)
```

Comment:
This observer never seems to get removed.

Reply:
That's intentional. As of iOS 8 you only remove them manually if you want to stop observing early. See the discussion in https://developer.apple.com/documentation/foundation/notificationcenter/1415360-addobserver.